feat: upgrade dockerfiles to be compatible with python 3.8.0 to 3.12.2

Faraz32123 opened the pull request:
close #68

Commit:
feat: upgrade dockerfiles to be compatible with 3.12.
2

DawoudSheraz commented on tutorecommerce/templates/ecommerce/build/ecommerce-worker/Dockerfile:
```diff
@@ -15,14 +16,25 @@ RUN mkdir /openedx/ecommerce_worker && \
+# https://github.com/pyenv/pyenv/releases
+ARG PYTHON_VERSION=3.12.2
+ENV PYENV_ROOT /opt/pyenv
+USER root
+RUN git clone https://github.com/pyenv/pyenv $PYENV_ROOT --branch v2.3.36 --depth 1
+# Install Python
+RUN $PYENV_ROOT/bin/pyenv install $PYTHON_VERSION
+USER app
```
why do we need to switch users here?

Faraz32123 replied:
```
 => [minimal  7/19] RUN echo "{}" > /openedx/config.yml                                                                          0.2s
 => ERROR [minimal  8/19] RUN git clone https://github.com/pyenv/pyenv /opt/pyenv --branch v2.3.36 --depth 1                     0.4s
------                                                                                                                                
 > importing cache manifest from docker.io/overhangio/openedx-discovery:17.0.0-nightly-cache:
------
------
 > [minimal  8/19] RUN git clone https://github.com/pyenv/pyenv /opt/pyenv --branch v2.3.36 --depth 1:
0.341 fatal: could not create work tree dir '/opt/pyenv': Permission denied
------
Dockerfile:36
--------------------
  34 |     ENV PYENV_ROOT /opt/pyenv
  35 |     # USER root
  36 | >>> RUN git clone https://github.com/pyenv/pyenv $PYENV_ROOT --branch v2.3.36 --depth 1
  37 |     # Install Python
  38 |     RUN $PYENV_ROOT/bin/pyenv install $PYTHON_VERSION
--------------------
ERROR: failed to solve: process "/bin/sh -c git clone https://github.com/pyenv/pyenv $PYENV_ROOT --branch v2.3.36 --depth 1" did not complete successfully: exit code: 128
```

DawoudSheraz replied:
👍🏽 nit: we can add that as a comment in file that unless switched, we get permission denied. 

But if change user, dont we need to give app user permissions on the repo cloned by root?

Faraz32123 replied:
added a comment.
nope we don't need that.

Commit:
docs: add a comment explaining the need of root user
 => [minimal  7/19] RUN echo "{}" > /openedx/config.yml                                                                          0.2s
 => ERROR [minimal  8/19] RUN git clone https://github.com/pyenv/pyenv /opt/pyenv --branch v2.3.36 --depth 1                     0.4s
------
 > importing cache manifest from docker.io/overhangio/openedx-discovery:17.0.0-nightly-cache:
------
------
 > [minimal  8/19] RUN git clone https://github.com/pyenv/pyenv /opt/pyenv --branch v2.3.36 --depth 1:
0.341 fatal: could not create work tree dir '/opt/pyenv': Permission denied
------
Dockerfile:36
--------------------
  34 |     ENV PYENV_ROOT /opt/pyenv
  35 |     # USER root
  36 | >>> RUN git clone https://github.com/pyenv/pyenv $PYENV_ROOT --branch v2.3.36 --depth 1
  37 |     # Install Python
  38 |     RUN $PYENV_ROOT/bin/pyenv install $PYTHON_VERSION
--------------------
ERROR: failed to solve: process "/bin/sh -c git clone https://github.com/pyenv/pyenv $PYENV_ROOT --branch v2.3.36 --depth 1" did not complete successfully: exit code: 128

regisb commented:
Does this mean that ecommerce is working with python 3.12? It would be absolutely fantastic, because we don't have anyone in the community to work on upgrading this IDA https://discuss.openedx.org/t/these-idas-will-not-be-upgraded-to-python-3-11-by-redwood/12755

Commit:
change ecommerce-worker dockerfile according to celery version 5.3.6

Faraz32123 commented:
> Does this mean that ecommerce is working with python 3.12? It would be absolutely fantastic, because we don't have anyone in the community to work on upgrading this IDA https://discuss.openedx.org/t/these-idas-will-not-be-upgraded-to-python-3-11-by-redwood/12755

Yes, ecommerce is working using these PRs, [PR for ecommerce](https://github.com/openedx/ecommerce/pull/4147) and [PR for ecommerce-worker](https://github.com/openedx/ecommerce-worker/pull/247).
But still we need to update github workflows for ecommerce's PR as the tests are failing.

Faraz32123 commented:
moved this PR to blocked as this PR cannot be merged until [PR for ecommerce](https://github.com/openedx/ecommerce/pull/4147) and [PR for ecommerce-worker](https://github.com/openedx/ecommerce-worker/pull/247) didn't merge.

Commit:
add changelog entry
